src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -298,6 +298,11 @@ private BoundForEachStatement BindForEachPartsWorker(BindingDiagnosticBag diagno
                         SourceLocalSymbol local = this.IterationVariable;
                         local.SetTypeWithAnnotations(declType);
 
+                        if (CheckRestrictedTypeInAsyncMethod(this.ContainingMemberOrLambda, declType.Type, diagnostics, typeSyntax))
+                        {
+                            hasErrors = true;
```

Comment:
>hasErrors = true; [](http://example.com/codeflow?start=28&length=17)

What is our intent here. We don't need to record every possible error in a bound node. The only requirement to set HasError flag is when the node is completely broken. I do not it is the case here. We were able to bind all things.

Reply:
Setting HasError without a good reason is likely to disable some useful analysis

Reply:
This was out of consistency with other callers of `CheckRestrictedTypeInAsyncMethod`. But I agree this error doesn't prevent us from binding.